src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
```

Comment:
The use of `Objects.equals` is against the checklist rule: 'Don't use Objects, Arrays, or any other util class.' Consider implementing your own equality check.